### PR TITLE
Fix not stopping correctly on failed power flow if configured to stop

### LINKED ISSUE #800: Stop simulation at powerflow exception

### LINKED ISSUE #815: Enhace Newton Raphson PowerFlow failure with more information
Currently the newton raphson powerflow fails without giving the user a proper cause.

### PR DESCRIPTION
**Pull request by staudtMarius**
Resolves #800 
Resolves #815

### CODE REVIEW

**Review by t-ober — APPROVED** [review]

File: src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala [hunk 1]
```diff
@@ -404,7 +407,16 @@ trait DBFSAlgorithm extends PowerFlowSupport with GridResultsSupport {
...
+                case failedResult: FailedNewtonRaphsonPFResult =>
+                  if (failedResult.cause == MaxIterationsReached) {
+                    ctx.log.warn(
+                      "Power flow in grid {} failed after {} iterations. Cause: {}",
+                      ctx.self,
+                      failedResult.iteration,
+                      failedResult.cause,
+                    )
+                  }
+
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala:419]: I am not really sure if this is what we want. If I recall correctly, `FailedPowerFlow` is supposed to "bubble up" towards the slack GridAgent, who then decides to quit DBFS simulation. But it looks like the information is not forwarded in `goToPowerFlowCalculationOrStay`.

So if everything actually worked, I think only the slack GridAgent should print out a warning that some subgrid has failed powerflow calculation. Additionally each GA in which Newton-Raphson actually failed, could print out a warning as well.

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Could you enhance `DBFSAlgorithmFailedPowerFlowSpec` with a test case where the inferior grid sends `FailedPowerFlow` to the center agent, which is then supposed to send the same to the superior agent.

File: src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala [hunk 2]
```diff
@@ -415,7 +427,7 @@ trait DBFSAlgorithm extends PowerFlowSupport with GridResultsSupport {
               )
 
               sender ! FailedPowerFlow
-              Behaviors.same
+              simulateGrid(gridAgentBaseData, currentTick)
```
[the remark this answers is not in the document]
Comment (staudtMarius) [reply src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala:430]: I don't know. I adapted this part to be the same as the other case above.

**Comment by staudtMarius** [comment]
!test

File: src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala:418]: Looked into it once more, in this case we can just stop, actually.

**Pull request merged**